### PR TITLE
feat(admin): add batch updating

### LINKED ISSUE #109: Update translations
Hello! This plugin batch translation works fine only when you have created new item.
I need batch translation of updated items.  
It look's like updated items should be marked somehow to be avaliable for batch update of previously batch translated languages.

### PR DESCRIPTION
**Pull request by layaxx**
listens for changes to localized entities and offers batch translation for updated entities

fix #109

### CODE REVIEW

File: plugin/server/services/updated-entry.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        results: [firstResult],
+      } = await super.find({
+        fields: ['id'],
+        filters: {
+          contentType: params.data.contentType,
+          groupID: params.data.groupID,
+        },
+      })
+      if (firstResult) {
+        return super.update(firstResult.id, {
```
Comment (sargreal) [inline plugin/server/services/updated-entry.js:18]: I am always getting this error when updating an entity:

```
[2023-04-02 23:55:38.930] http: PUT /content-manager/collection-types/api::article.article/1014 (43 ms) 200
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
ValidationError: Invalid payload submitted for the update of an entity of type Translate updated Entry. Expected an object, but got undefined
    at Object.validateEntityUpdate (/home/felix/Documents/coding/strapi-develop/translate/node_modules/@strapi/strapi/lib/services/entity-validator/index.js:230:13)
    at Object.update (/home/felix/Documents/coding/strapi-develop/translate/node_modules/@strapi/strapi/lib/services/entity-service/index.js:184:45)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.<anonymous> (/home/felix/Documents/coding/strapi-develop/translate/node_modules/@strapi/strapi/lib/services/entity-service/index.js:327:20
```

File: plugin/admin/src/components/BatchUpdateTable/Table.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            </Th>
+            {locales.map((locale) => (
+              <Th key={locale.code}>
+                <Typography variant="sigma">{locale.name}</Typography>
+              </Th>
+            ))}
+          </Tr>
+        </Thead>
+        <Tbody>
+          {updates.map(({ id, attributes }, index) => (
```
Comment (sargreal) [inline plugin/admin/src/components/BatchUpdateTable/Table.js:125]: I feel like this table becomes unreadable when many entites are updated. Would there be a possiblity to group these updates by content type?

Optimally I would even like to have only One table, so would it be possible to integrate the functionality into the existing table?

File: plugin/server/services/translate.js [hunk 3]
```diff
@@ -69,6 +75,85 @@ module.exports = ({ strapi }) => ({
...
+          contentTypeSchema
+        )
+
+        console.log({
+          translated,
+          uidsUpdated,
+          withFieldsDeleted,
+          fullyTranslatedData,
+          val: strapi.config.get('plugin.translate').regenerateUids,
+        })
```
Comment (sargreal) [inline plugin/server/services/translate.js:145]: Forgot a console.log
```suggestion
```

File: plugin/server/services/translate.js [hunk 4]
```diff
@@ -69,6 +75,85 @@ module.exports = ({ strapi }) => ({
   async batchTranslateCancelJob(id) {
     return this.batchTranslateManager.cancelJob(id)
   },
+  async batchUpdate(params) {
```
Comment (sargreal) [inline plugin/server/services/translate.js:78]: I believe doing this synchronously is not a good Idea as the translation might take a long time. 

I believe integrating this functionality into the BatchTranslateJob would be more coherent, but would probably require some refactoring.

**Review by sargreal — CHANGES_REQUESTED** [review]
The basic functionality looks good, however I think it does not feel coherent with the rest of the application. I think it would make more sense to integrate the batch translation and the batch translation updates with each other.

I think this would also make it easier to add a possible automatic hooked translation/updates later

File: plugin/server/services/updated-entry.js (hunk 1 above)
Comment (layaxx) [reply plugin/server/services/updated-entry.js:18]: will look into this

File: plugin/admin/src/components/BatchUpdateTable/Table.js (hunk 2 above)
Comment (layaxx) [reply plugin/admin/src/components/BatchUpdateTable/Table.js]: I think the way it is implemented atm, it should be in a separate table. Unlike the batch translations, which works on a content type level (translating every entity of a given content type from one source locale to one target locale), batch updating works on an entity level (all translations of a given entity are retranslated from a given source locale). 

By default, there is now only one row per content type, which can be expanded to show all entries.

File: plugin/admin/src/components/BatchUpdateTable/Table.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+              key={entry.id}
+              entry={entry}
+              selectedIDs={selectedUpdateIDs}
+              setSelectedIDs={setSelectedUpdateIDs}
+              locales={locales}
+              index={index}
+            />
+          ))}
+        </Tbody>
+      </Table>
```
Comment (sargreal) [inline plugin/admin/src/components/BatchUpdateTable/Table.js:64]: Should have some margin bottom 

Props: https://github.com/strapi/design-system/blob/main/packages/strapi-design-system/src/Box/Box.tsx

**Review by sargreal — CHANGES_REQUESTED** [review]
The update modal seems a bit too crowded, there should be some margins etc.

Also the messages shown are a bit confusing, these should be different from the messages for the direct translation right?

image

**Review by sargreal — APPROVED** [review]
LGTM

**Pull request merged**